adapters/bidder.go

```diff
@@ -48,10 +48,12 @@ type ResponseData struct {
 
 // RequestData packages together the fields needed to make an http.Request.
 type RequestData struct {
-	Method  string
-	Uri     string
-	Body    []byte
-	Headers http.Header
+	Method       string
+	Uri          string
+	Body         []byte
+	Headers      http.Header
+	AuthUserName string      `json:",omitempty"`
+	AuthPassword string      `json:",omitempty"`
```

Comment:
I think we can find a better way to handle this.

As written, suppose a Bidder returns `Headers` which include `Authorization`, `AuthUserName`, _and_ `AuthPassword`. Which one should exist in the outgoing request?

I don't see a clear answer... which means there's no way to follow the "principle of least surprise".

Instead, I'd recommend writing a `func (r *RequestData) SetBasicAuth(username string, password string)` function which edits the `Headers` directly. You could basically just copy the implementation from the Go core:

```
r.Headers.Set("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(username + ":" + password))
```

It'd be equally convenient for Bidders, but the behavior would be much more clear.

Reply:
good point. I agree with creating a func for it. will add this.